prymer/api/primer_pair.py

```diff
@@ -226,29 +230,37 @@ def __str__(self) -> str:
             + f"{self.amplicon_tm}\t{self.penalty}"
         )
 
-    def _calculate_amplicon(self) -> Span:
+    @staticmethod
+    def calculate_amplicon_span(left_primer: Oligo, right_primer: Oligo) -> Span:
         """
-        Calculates the amplicon from the left and right primers, spanning from the start of the
-        left primer to the end of the right primer.
+        Calculates the amplicon Span from the left and right primers.
+
+        Args:
+            left_primer: the left primer for the amplicon
+            right_primer: the right primer for the amplicon
+
+        Returns:
+            a Span starting at the first base of the left primer and ending at the last base of
+             the right primer
```

Comment:
**suggestion** Enumerate the cases where this function will raise an exception.

```suggestion
            A Span starting at the first base of the left primer and ending at the last base of
            the right primer.
            
        Raises:
            ValueError: If `left_primer` and `right_primer` have different reference names.
            ValueError: If `left_primer` is not to the left of`right_primer`. The two primers may overlap, but the left primer must start before the first base of the right primer, and end before the last base of the right primer.
```

(NB: Please run ruff to wrap the line above)

Reply:
**question** I've written the docstring above to reflect the behavior of the current implementation, but _are_ the primers allowed to overlap?

Reply:
It would be weird, because the more they overlap the more you're likely to make primer dimer, but I see no reason the code should forbid it.